### ISSUE TITLE
Background url without quotes fails

### ISSUE THREAD

**Issue by kallaspriit** [TrySound/postcss-value-parser#1]
Just today my build server began to fail with:
```
error	25-Aug-2015 14:36:27	ERROR in ./gfx/css/main.css
error	25-Aug-2015 14:36:27	Module build failed: ModuleBuildError: Module build failed: TypeError: Cannot read property 'type' of undefined
error	25-Aug-2015 14:36:27	    at module.exports (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/cssnano/node_modules/postcss-convert-values/node_modules/postcss-value-parser/lib/tokenize.js:62:24)
error	25-Aug-2015 14:36:27	    at new Parser (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/cssnano/node_modules/postcss-convert-values/node_modules/postcss-value-parser/lib/parser.js:5:18)
error	25-Aug-2015 14:36:27	    at module.exports (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/cssnano/node_modules/postcss-convert-values/node_modules/postcss-value-parser/lib/parser.js:50:12)
error	25-Aug-2015 14:36:27	    at optimise (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/cssnano/node_modules/postcss-convert-values/dist/index.js:26:54)
error	25-Aug-2015 14:36:27	    at /opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:117:34
error	25-Aug-2015 14:36:27	    at /opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:102:26
error	25-Aug-2015 14:36:27	    at Rule.each (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:89:22)
error	25-Aug-2015 14:36:27	    at Rule.eachInside (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:101:21)
error	25-Aug-2015 14:36:27	    at /opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:105:32
error	25-Aug-2015 14:36:27	    at Root.each (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:89:22)
error	25-Aug-2015 14:36:27	    at Root.eachInside (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:101:21)
error	25-Aug-2015 14:36:27	    at Root.eachDecl (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:115:25)
error	25-Aug-2015 14:36:27	    at /opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/cssnano/node_modules/postcss-convert-values/dist/index.js:54:13
error	25-Aug-2015 14:36:27	    at LazyResult.run (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/lazy-result.js:197:24)
error	25-Aug-2015 14:36:27	    at LazyResult.sync (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/lazy-result.js:183:32)
error	25-Aug-2015 14:36:27	    at LazyResult.stringify (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/lazy-result.js:214:14)
error	25-Aug-2015 14:36:27	    at DependenciesBlock.onModuleBuildFailed (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:315:19)
error	25-Aug-2015 14:36:27	    at nextLoader (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:270:31)
error	25-Aug-2015 14:36:27	    at /opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:292:15
error	25-Aug-2015 14:36:27	    at runSyncOrAsync (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:173:4)
error	25-Aug-2015 14:36:27	    at nextLoader (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:290:3)
error	25-Aug-2015 14:36:27	    at /opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:292:15
error	25-Aug-2015 14:36:27	    at context.callback (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:148:14)
error	25-Aug-2015 14:36:27	    at /opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/postcss-loader/index.js:46:13
error	25-Aug-2015 14:36:27	    at run (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/babel-core/node_modules/core-js/modules/es6.promise.js:89:39)
error	25-Aug-2015 14:36:27	    at /opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/babel-core/node_modules/core-js/modules/es6.promise.js:100:28
error	25-Aug-2015 14:36:27	    at flush (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/babel-core/node_modules/core-js/modules/$.microtask.js:9:13)
error	25-Aug-2015 14:36:27	    at process._tickCallback (node.js:415:13)
error	25-Aug-2015 14:36:27	 @ ./app.js 29:21-50
error	25-Aug-2015 14:36:27	Child extract-text-webpack-plugin:
error	25-Aug-2015 14:36:27	    
error	25-Aug-2015 14:36:27	    ERROR in ./~/css-loader!./~/postcss-loader!./gfx/css/main.css
error	25-Aug-2015 14:36:27	    Module build failed: TypeError: Cannot read property 'type' of undefined
error	25-Aug-2015 14:36:27	        at module.exports (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/cssnano/node_modules/postcss-convert-values/node_modules/postcss-value-parser/lib/tokenize.js:62:24)
error	25-Aug-2015 14:36:27	        at new Parser (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/cssnano/node_modules/postcss-convert-values/node_modules/postcss-value-parser/lib/parser.js:5:18)
error	25-Aug-2015 14:36:27	        at module.exports (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/cssnano/node_modules/postcss-convert-values/node_modules/postcss-value-parser/lib/parser.js:50:12)
error	25-Aug-2015 14:36:27	        at optimise (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/cssnano/node_modules/postcss-convert-values/dist/index.js:26:54)
error	25-Aug-2015 14:36:27	        at /opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:117:34
error	25-Aug-2015 14:36:27	        at /opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:102:26
error	25-Aug-2015 14:36:27	        at Rule.each (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:89:22)
error	25-Aug-2015 14:36:27	        at Rule.eachInside (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:101:21)
error	25-Aug-2015 14:36:27	        at /opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:105:32
error	25-Aug-2015 14:36:27	        at Root.each (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:89:22)
error	25-Aug-2015 14:36:27	        at Root.eachInside (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:101:21)
error	25-Aug-2015 14:36:27	        at Root.eachDecl (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/container.js:115:25)
error	25-Aug-2015 14:36:27	        at /opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/cssnano/node_modules/postcss-convert-values/dist/index.js:54:13
error	25-Aug-2015 14:36:27	        at LazyResult.run (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/lazy-result.js:197:24)
error	25-Aug-2015 14:36:27	        at LazyResult.sync (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/lazy-result.js:183:32)
error	25-Aug-2015 14:36:27	        at LazyResult.stringify (/opt/bamboo-agent-home/xml-data/build-dir/MCC-MCCFE-JOB1/node_modules/css-loader/node_modules/postcss/lib/lazy-result.js:214:14)
```

After some investigation the problem seems to be with having a background url path without quotes. For example this fails:
```
background: url(/gfx/img/bg.jpg) top left no-repeat;
```

Yet this is fine:
```
background: url("/gfx/img/bg.jpg") top left no-repeat;
```

**Comment by TrySound** [comment]
@kallaspriit Which version of `postcss-convert-values`? 1.3.0 was broken, but already fixed in 1.3.1.

**Comment by ben-eb** [comment]
I can reproduce. The error is in here & not convert-values.

Test case with convert-values:

```js
{
    message: 'should not crash on quoteless urls',
    fixture: 'h1{background: url(/gfx/img/bg.jpg) top left no-repeat}',
    expected: 'h1{background: url(/gfx/img/bg.jpg) top left no-repeat}'
}
```

Outputs:

```
/Users/beneb/projects/postcss-convert-values/node_modules/postcss/lib/lazy-result.js:200
            throw error;
                  ^
TypeError: Cannot read property 'type' of undefined
    at module.exports (/Users/beneb/projects/postcss-convert-values/node_modules/postcss-value-parser/lib/tokenize.js:62:24)
    at new Parser (/Users/beneb/projects/postcss-convert-values/node_modules/postcss-value-parser/lib/parser.js:5:18)
    at module.exports (/Users/beneb/projects/postcss-convert-values/node_modules/postcss-value-parser/lib/parser.js:50:12)
    at optimise (/Users/beneb/projects/postcss-convert-values/src/index.js:12:18)
    at /Users/beneb/projects/postcss-convert-values/node_modules/postcss/lib/container.js:117:34
    at /Users/beneb/projects/postcss-convert-values/node_modules/postcss/lib/container.js:102:26
    at Rule.each (/Users/beneb/projects/postcss-convert-values/node_modules/postcss/lib/container.js:89:22)
    at Rule.eachInside (/Users/beneb/projects/postcss-convert-values/node_modules/postcss/lib/container.js:101:21)
    at /Users/beneb/projects/postcss-convert-values/node_modules/postcss/lib/container.js:105:32
    at Root.each (/Users/beneb/projects/postcss-convert-values/node_modules/postcss/lib/container.js:89:22)
```
